File: super-duper-octo-palm-tree.app/models/Order.cs
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;

namespace super_duper_octo_palm_tree.app.models
{
    public class Order
    {
        public User User { get; set; }

        public int NbBought { get { return TicketList.Count; } }

        public List<Ticket> TicketList { get; set; }

        public double TotalBasePrice {
            get
            {
                return TicketList.Sum(ticket => ticket.BasePrice);
            }
        }

        public double TotalAdditionalPrice
        {
            get
            {
                return TicketList.Sum(ticket => ticket.AdditionalPrice);
            }
        }

        public double TotalDiscountedBasePrice
        {
            get
            {
                return TicketList.Sum(ticket => ticket.DiscountedBasePrice);
            }
        }

        public double TotalPrice {
            get {
                return TotalDiscountedBasePrice + TotalAdditionalPrice;
            }
        }

        //[JsonConverter(typeof(JsonStringEnumConverter))]
        public Currency UsedCurrency { get; set; }

        public double ExchangeRate { get; set; }

        public bool IsPaid { get; set; }
    }
}

File: App/src/models/order.ts
import { Ticket } from "./ticket";
import { User } from "./user";
import { Currency, CurrencyType } from "./currency";
import { FlightSource } from "./flightSource.enum";

export interface Order {
  user: User
  nbBought: number
  ticketList: Array<Ticket>
  date: string
  totalBasePrice: number
  totalAdditionalPrice: number
  totalDiscountedBasePrice: number
  totalPrice: number
  usedCurrency: CurrencyType
  exchangeRate: number
  isPaid: boolean
  flightSource: FlightSource;
  extraData?: any;
}